apps/whale-api/src/module.api/poolpair.controller.e2e.ts

```diff
@@ -513,11 +519,76 @@ describe('get best path', () => {
           poolPairId: '23',
           priceRatio: { ab: '0.14285714', ba: '7.00000000' },
           tokenA: { id: '10', symbol: 'J', displaySymbol: 'dJ' },
-          tokenB: { id: '11', symbol: 'K', displaySymbol: 'dK' }
+          tokenB: { id: '11', symbol: 'K', displaySymbol: 'dK' },
+          commissionFeeInPct: '0.25000000'
         }
       ],
       estimatedReturn: '7.00000000',
-      estimatedReturnLessDexFees: '7.00000000'
+      estimatedReturnLessDexFees: '5.25000000'
+    })
+  })
+
+  it('should deduct commission fee - 1 leg', async () => {
+    const response = await controller.getBestPath('10', '12')
+    expect(response).toStrictEqual({
+      fromToken: {
+        id: '10',
+        symbol: 'J',
+        displaySymbol: 'dJ'
+      },
+      toToken: {
+        id: '12',
+        symbol: 'L',
+        displaySymbol: 'dL'
+      },
+      bestPath: [
+        {
+          symbol: 'J-L',
+          poolPairId: '24',
+          priceRatio: { ab: '0.50000000', ba: '2.00000000' },
+          tokenA: { id: '10', symbol: 'J', displaySymbol: 'dJ' },
+          tokenB: { id: '12', symbol: 'L', displaySymbol: 'dL' },
+          commissionFeeInPct: '0.10000000'
+        }
+      ],
+      estimatedReturn: '2.00000000',
+      estimatedReturnLessDexFees: '1.80000000'
+    })
+  })
+
+  it('should deduct commission fees - 2 legs', async () => {
```

Comment:
Can we also include a case with dexFee + commissionFee combined?

Reply:
Updated! Added dex fees on L-M pair
![image](https://user-images.githubusercontent.com/13292662/188809410-168f45c7-6c3d-4b31-b1f6-222f8e5bd9e3.png)